src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -61,9 +61,12 @@
 (handlers/register-handler-fx
   ::transaction-modal-completed
   (fn [{db :db} [_ {:keys [id response]}]]
-    (let [{:keys [hash error]} response]
-      (when-not (and error (string? error) (not (string/blank? error)))
-        {:db       (-> db
+    (let [{:keys [hash error]} response
+          db' (assoc-in db [:wallet/send-transaction :in-progress?] false)
+          has-error? (and error (string? error) (not (string/blank? error)))]
+      (if has-error?
+        {:db db'}
+        {:db       (-> db'
                        (update-in [:wallet :transactions-unsigned] dissoc id)
```

Comment:
```(update :wallet/send-transaction assoc :transaction-id nil  :wrong-password? false :in-progress? false)```

Reply:
This doesn't address the dissoc, which is the only reason this is threaded.

Reply:
@flexsurfer Is that fine for you?

Reply:
i meant why do we need `db' (assoc-in db [:wallet/send-transaction :in-progress?] false)` ? and why you don't have `assoc` in `update`?

Reply:
so i suggested
```
{:db       (-> db
                     (update-in [:wallet :transactions-unsigned] dissoc id)		                        
                     (update :wallet/send-transaction assoc :transaction-id nil :wrong-password? false :in-progress? false)

Reply:
right because there is `if` ,sorry